secret-provider/deploy/deploy.sh

```diff
@@ -33,10 +34,17 @@ deploy() {
 }
 
 case "$1" in
-undeploy)
+  deploy)
+    deploy
```

Comment:
I wasn't sure where to put this comment. But while working on some other local testing I noticed that the secret-provider has its own service account. When using your own images this service account does not have the imagePullSecret added. 
So we should either add that here or in `registry/deploy.sh`

```
kubectl patch serviceaccount secret-provider -p \
            "{\"imagePullSecrets\": [{\"name\": \"$NAME\"}]}" \
            -n $namespace
```

Reply:
I'm not sure I follow, why does it need `imagePullSecret`? I thought now the image comes from a public registry